components/tidb_query/src/codec/convert.rs

```diff
@@ -284,159 +269,190 @@ impl ToInt for Real {
+        let s = get_valid_utf8_prefix(ctx, self)?;
+        s.convert(ctx)
     }
 }
 
-impl ToInt for Decimal {
-    #[inline]
-    fn to_int(&self, ctx: &mut EvalContext, tp: FieldTypeTp) -> Result<i64> {
-        // TODO: avoid this clone
-        let dec = round_decimal_with_ctx(ctx, self.clone())?;
-        let val = dec.as_i64_with_ctx(ctx)?;
-        val.to_int(ctx, tp)
+impl ConvertTo<i64> for &str {
+    /// Port from TiDB's types.StrToInt
+    fn convert(&self, ctx: &mut EvalContext) -> Result<i64> {
+        let s = self.trim();
+        let s = get_valid_int_prefix(ctx, s)?;
+        match s.parse::<i64>() {
+            Ok(val) => Ok(val),
+            Err(_) => {
+                ctx.handle_overflow_err(Error::overflow("BIGINT", s.to_string()))?;
+                // in TiDB, the strconv.ParseInt return i64::MAX or i64::MIN when overflow,
+                // according to  strconv.ParseInt's doc:
+                // if the value corresponding to s cannot be represented by a signed integer of the given size,
+                // the returned value is the maximum magnitude integer of the appropriate bitSize and sign
+                if s.starts_with('-') {
+                    Ok(integer_signed_lower_bound(FieldTypeTp::LongLong))
+                } else {
+                    Ok(integer_signed_upper_bound(FieldTypeTp::LongLong))
+                }
+            }
+        }
     }
+}
 
-    #[inline]
-    fn to_uint(&self, ctx: &mut EvalContext, tp: FieldTypeTp) -> Result<u64> {
-        // TODO: avoid this clone
-        let dec = round_decimal_with_ctx(ctx, self.clone())?;
-        decimal_as_u64(ctx, dec, tp)
+impl ConvertTo<u64> for &str {
+    /// Port from TiDB's types.StrToUint
+    fn convert(&self, ctx: &mut EvalContext) -> Result<u64> {
+        let s = self.trim();
+        let s = get_valid_int_prefix(ctx, s)?;
+        match s.parse::<u64>() {
+            Ok(v) => Ok(v),
+            Err(_) => {
+                ctx.handle_overflow_err(Error::overflow("BIGINT UNSIGNED", s))?;
+                // in TiDB, the strconv.ParseUint return u64::MAX when non-negative number overflow,
+                // according to  strconv.ParseInt's doc:
+                // if the value corresponding to s cannot be represented by a signed integer of the given size,
+                // the returned value is the maximum magnitude integer of the appropriate bitSize and sign
+                Ok(integer_unsigned_upper_bound(FieldTypeTp::LongLong))
+            }
+        }
     }
 }
 
-impl ToInt for DateTime {
-    #[inline]
-    fn to_int(&self, ctx: &mut EvalContext, tp: FieldTypeTp) -> Result<i64> {
-        // TODO: avoid this clone after refactor the `Time`
-        let mut t = self.clone();
-        t.round_frac(DEFAULT_FSP)?;
-        let dec: Decimal = t.convert(ctx)?;
-        let val = dec.as_i64_with_ctx(ctx)?;
-        val.to_int(ctx, tp)
+impl ConvertTo<i64> for Bytes {
+    /// Port from TiDB's types.StrToInt
```

Comment:
It's a delegate implementation for `impl ConvertTo<i64> for &str`, so I think it's better to keep this comment in  `impl ConvertTo<i64> for &str` and remove from those delegate implementations.